pandas/tests/tseries/offsets/test_offsets_properties.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+"""
+Behavioral based tests for offsets and date_range.
+
+This file is adapted from https://github.com/pandas-dev/pandas/pull/18761 -
+which was more ambitious but less idiomatic in its use of Hypothesis.
+
+You may wish to consult the previous version for inspiration on further
+tests, or when trying to pin down the bugs exposed by the tests below.
+"""
+
+import pytest
+from hypothesis import given, assume, strategies as st
+from hypothesis.extra.pytz import timezones as pytz_timezones
+from hypothesis.extra.dateutil import timezones as dateutil_timezones
+
+import pandas as pd
+
+from pandas.tseries.offsets import (
+    Hour, Minute, Second, Milli, Micro, Nano,
```

Comment:
these already exist as fixtures, i'd like to avoid specifically importing things like this and use the fixture / generating fixture instead

Reply:
This is actually a non-trivial trade-off!  Because we can't call fixtures outside of a test, we'll still need to import these types to register the strategy for each of them in `conftest.py` (see next comment).

In the test file, we have two options:
1. Import them, and let Hypothesis decide which to try.
2. Use a fixture which creates a test for each, and have Hypothesis try cases for each independently.

(1) is faster at runtime because Hypothesis is running one instead of *n* tests, while (2) is more explicit if a little less elegant.  Your call.